roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,32 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set slrouter address
+  set_fact: sl_router_address="{{ ansible_default_ipv4.network.split('.')[0] }}.{{ ansible_default_ipv4.network.split('.')[1] }}.{{ansible_default_ipv4.network.split('.')[2] }}.{{ (ansible_default_ipv4.network.split('.')[3] | int) + 1 }}"
+  when: ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device   
+
+- name: add static route for multi subnet
+  command: "/sbin/ip route replace {{ item }} via {{ sl_router_address }}"
+  when: (not (ansible_default_ipv4.network ==  ( item | ipaddr('network') ))) and (ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device)
+  with_items: "{{ undercloud_cidr }}"
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces"
+  when: network_interfaces is defined
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg"
+  when: network_interfaces_d is defined
+
+- name: Check for static routes in {{ nw_config_file }}
+  command: "grep -h 'ip route add {{ item }} via {{ sl_router_address }}' {{ nw_config_file }}"
+  with_items: "{{ undercloud_cidr }}"
+  always_run: True
+  changed_when: False
+  register: routes_defined
+  failed_when: routes_defined.rc != 0
+  when: (not ( ansible_default_ipv4.network  == ( item | ipaddr('network') ))) and (ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device)
```

Comment:
If you find yourself using `hostvars[inventory_hostname]`, you can typically leave off that part. Variable lookups start at the current host being worked on, so `hostvars[inventory_hostname][primary_interface].device` could be referenced as `primary_interface.device`.